packages/subapp-react/test/spec/fe-framework.spec.js

```diff
@@ -62,4 +62,58 @@ describe("FE React framework", function() {
     const c = framework.renderStart();
     expect(c.type).equals(Component);
   });
+
+  it("should render and unmount component into DOM element", () => {
+    const dom = new JSDOM(`<div id="test"><p>Hello bar</p></div>`);
+    global.window = dom.window;
+    const element = dom.window.document.getElementById("test");
+    const framework = new feLib.FrameworkLib({
+      subApp: {
+        info: {
+          Component: props => <p>Hello {props.foo}</p>,
+          subappRoot: {}
+        }
+      },
+      element,
+      options: { props: { foo: "bar" } }
+    });
+
+    framework.renderStart();
+    const root = framework.ref.subApp.info.subappRoot;
+
+    expect(element.innerHTML).to.equal(`<p>Hello bar</p>`);
+
+    act(() => {
+      root.unmount();
+    });
+
+    expect(element.innerHTML).to.equal("");
+  });
+
+  it("should hydrate and unmount a server-side rendered component", () => {
+    const dom = new JSDOM(`<div id="test"><p>Hello bar</p></div>`);
+    global.window = dom.window;
+    const element = dom.window.document.getElementById("test");
+    const framework = new feLib.FrameworkLib({
+      subApp: {
+        info: {
+          Component: props => <p>Hello {props.foo}</p>,
+          subappRoot: {}
+        }
+      },
+      element,
+      options: { props: { foo: "bar" }, serverSideRendering: true }
+    });
+
+    framework.renderStart();
+    const root = framework.ref.subApp.info.subappRoot;
+
+    expect(element.innerHTML).to.equal("<p>Hello bar</p>");
+
+    act(() => {
+      root.unmount();
+    });
+
+    expect(element.innerHTML).to.equal("");
+  });
```

Comment:
Does it make sense to add a use case where the server content is mismatched from the client's content?

https://react.dev/reference/react-dom/client/hydrateRoot#caveats

Reply:
@kylejeske Hydration itself is already implemented, so probably that should be on a different PR. 
This specifically was to test the enhancement to expose root and then app could do some lifecycle management. (line 114)

Reply:
Also, if there is a mismatch from server content, react itself would show warning and application need to fix that first before they use root for unmount IMO.